controls/apigateway.sp

```diff
@@ -67,3 +73,51 @@ control "apigateway_stage_logging_enabled" {
     soc_2              = "true"
   })
 }
+
+control "apigateway_rest_api_create_before_destroy_enabled" {
+  title       = "API Gateway REST API should have create_before_destroy enabled"
+  description = "This control checks whether AWS API Gateway REST API has create_before_destroy enabled. It is recommended to enable the resource lifecycle configuration block create_before_destroy argument in this resource configuration to manage all requests that use this API, avoiding an outage."
+  query       = query.apigateway_rest_api_create_before_destroy_enabled
+
+  tags = local.apigateway_compliance_common_tags
+}
+
+control "apigateway_deployment_create_before_destroy_enabled" {
+  title       = "API Gateway Deployment should have create_before_destroy enabled"
+  description = "This control checks whether AWS API Gateway Deployment has create_before_destroy enabled. It is recommended to enable the resource lifecycle configuration block create_before_destroy argument in this resource configuration to manage all requests that use this API, avoiding an outage."
+  query       = query.apigateway_deployment_create_before_destroy_enabled
+
+  tags = local.apigateway_compliance_common_tags
+}
+
+control "apigateway_method_settings_cache_enabled" {
+  title       = "API Gateway Method Settings should have cache enabled"
+  description = "This control checks whether AWS API Gateway Method Settings has cache enabled. It is recommended to enable cache for all methods in API Gateway."
+  query       = query.apigateway_method_settings_cache_enabled
+
+  tags = local.apigateway_compliance_common_tags
+}
+
+control "apigateway_method_settings_cache_encryption_enabled" {
+  title       = "API Gateway Method Settings should have cache encrypted"
+  description = "This control checks whether AWS API Gateway Method Settings has cache encrypted. It is recommended to enable cache encryption for all methods in API Gateway."
+  query       = query.apigateway_method_settings_cache_encryption_enabled
+
+  tags = local.apigateway_compliance_common_tags
+}
+
+control "apigateway_method_settings_data_trace_enabled" {
+  title       = "API Gateway Method Settings should have data trace enabled"
+  description = "This control checks whether AWS API Gateway Method Settings has data trace enabled. It is recommended to disable data trace for all methods in API Gateway."
```

Comment:
What is the query logic here? Should it be enabled or disabled? The query prompts it to be disabled while the title and description are contradicting.
I believe it should be `enabled` @Priyanka-Chatterjee-2000 please update it accordingly.

Reply:
No it should be disabled only.